src/middlewared/middlewared/etc_files/rc.conf.py

```diff
@@ -92,6 +92,18 @@ def kbdmap_config(middleware, context):
         return []
 
 
+def lldp_config(middleware, context):
+    lldp = middleware.call_sync('lldp.config')
+    ladvd_flags = ['-a']
+    if lldp['intdesc']:
+        ladvd_flags.append('-z')
+    if lldp['country']:
+        ladvd_flags += ['-c', lldp['country']]
+    if lldp['location']:
+        ladvd_flags += ['-l', rf'\"{lldp["location"]}\"']
```

Comment:
One of my dreams about `rc.conf` rewrite was using a proper quote method instead of just wrapping strings with `\"` because they can contain `"`, `$` or god know what

Reply:
What if we allow these methods to return this:

```
def f():
    yield {"service_enable": True}
    yield {"service_name": 'Transgalactic spaceship "FreeNAS 3000"'},
    yield {"service_flags": ["-p", "quoted parameter"]}
```

and it will automatically quote it and write

```
service_enable="YES"
service_name="Transgalactic spaceship \"FreeNAS 3000\""
service_flags="-p \"quoted parameter\""
```

I only recall quoting problems with minio so far, let's not have any more

Reply:
I thought about something like that but I was already mostly done.
And I do like the idea but do you think we can handle that after we get this merged?
I already spent a good chunk of time testing the code as is and its not really a regression but a nice to have improvement.

I am also hoping most of what is in rc.conf.py now is temporary and will be removed once we rewrite services management ; )

Reply:
I agree with most of what you say, but I think we still should handle quoting for variables that can have arbitrary symbols in them.